src/modules/endpoint/__tests__/DatabaseConnector.spec.js

```diff
@@ -0,0 +1,308 @@
+    })
+  })
+
+  describe('loadLanguages', () => {
+    it('should throw error if languages are not persisted', () => {
+      const context = new DatabaseContext('tcc', 'de')
+      expect(databaseConnector.loadLanguages(context)).rejects.toThrowError()
+    })
+    it('should return a value that matches the one that was stored', async () => {
+      const context = new DatabaseContext('tcc', 'de')
+      await databaseConnector.storeLanguages([testLanguage], context)
+
+      return databaseConnector.loadLanguages(context).then(languages => {
+        expect(languages).toEqual([testLanguage])
+      })
+    })
+  })
+
+  describe('isEventsPersisted', () => {
+    it('should return false if events are not persisted', () => {
+      const context = new DatabaseContext('tcc', 'de')
+      return databaseConnector.isEventsPersisted(context).then(isPersisted => {
+        expect(isPersisted).toBe(false)
+      })
+    })
+    it('should return true if events are persisted', async () => {
+      const context = new DatabaseContext('tcc', 'de')
+      await databaseConnector.storeEvents([testEvent], context)
+      return databaseConnector.isEventsPersisted(context).then(isPersisted => {
+        expect(isPersisted).toBe(true)
+      })
+    })
+  })
+
+  describe('storeEvents', () => {
+    it('should throw error if data is empty', () => {
+      const context = new DatabaseContext('tcc', 'de')
+      expect(databaseConnector.storeEvents([null], context)).rejects.toThrowError()
+    })
+  })
+
+  describe('loadEvents', () => {
+    it('should throw error if events are not persisted', () => {
+      const context = new DatabaseContext('tcc', 'de')
+      expect(databaseConnector.loadEvents(context)).rejects.toThrowError()
+    })
+    it('should return a value that matches the one that was stored', async () => {
+      const context = new DatabaseContext('tcc', 'de')
+      await databaseConnector.storeEvents([testEvent], context)
+
+      return databaseConnector.loadEvents(context).then(events => {
+        expect(events).toEqual([testEvent])
+      })
+    })
+  })
+
+  describe('storeResourceCache', () => {
+    it('should throw error if data is empty', () => {
+      const context = new DatabaseContext('tcc', 'de')
+      expect(databaseConnector.storeResourceCache({}, context)).rejects.toThrowError()
+    })
+  })
+
+  describe('loadResourceCache', () => {
+    it('should return an empty value if resources are not persisted', () => {
+      const context = new DatabaseContext('tcc', 'de')
+      return databaseConnector.loadResourceCache(context).then(cache => {
+        expect(cache).toEqual({})
+      })
+    })
+    it('should return a value that matches the one that was stored', async () => {
+      const context = new DatabaseContext('tcc', 'de')
+      await databaseConnector.storeResourceCache(testResources, context)
+
+      return databaseConnector.loadResourceCache(context).then(cache => {
+        expect(cache).toMatchSnapshot()
```

Comment:
I think testing this without the snapshots like you did in the other tests is better. I think `testResources` should be the same as the `cache`

Reply:
If I do it without snapshot testing, then there is a problem with lastUpdate (lastUpdate is a moment object in the expected data, but a string in the received data)

Reply:
Right, that is actually a bug right now. It will be resolved by the mapping for resource cache: https://issues.integreat-app.de/browse/NATIVE-330

Go with this such that the snapshot test doesn't stay
```suggestion
expect(cache).toEqual({
        'de': {
          '/path/to/page': {
            'https://test.de/path/to/resource/test.png': {
              'filePath': '/local/path/to/resource/b4b5dca65e423.png',
              'hash': 'testHash',
              'lastUpdate': '2011-02-04T23:00:00.000Z'
            }
          },
          '/path/to/page/child': {
            'https://test.de/path/to/resource/test2.jpg': {
              'filePath': '/local/path/to/resource/970c65c41eac0.jpg',
              'hash': 'testHash',
              'lastUpdate': '2011-02-04T23:00:00.000Z'
            }
          }
        }
      })
```

Reply:
Maybe also add a comment